src/parser/json/relationship_parser.py

```diff
@@ -0,0 +1,197 @@
+                                             default=[]))
+
+            file_dicts: List[Dict] = input_doc_dict.get("files")
+            if file_dicts:
+                # not implemented yet, deal with deprecated fields in file
+                relationships_list.extend(
+                    parse_field_or_log_error(logger=self.logger, field=file_dicts,
+                                             parsing_method=self.parse_file_dependencies, default=[]))
+
+            generated_relationships = self.parse_artifact_of(file_dicts=file_dicts)
+
+            raise_parsing_error_if_logger_has_messages(self.logger)
+
+        return relationships_list
+
+    def parse_relationships(self, relationship_dicts: List[Dict]) -> List[Relationship]:
+        logger = Logger()
+        relationship_list = []
+        for relationship_dict in relationship_dicts:
+            relationship_list = append_parsed_field_or_log_error(logger=logger, list_to_append_to=relationship_list,
+                                                                 field=relationship_dict, method_to_parse=self.parse_relationship)
+        raise_parsing_error_if_logger_has_messages(logger)
+        return relationship_list
+
+    def parse_relationship(self, relationship_dict: Dict) -> Relationship:
+        logger = Logger()
+        spdx_element_id: str = relationship_dict.get("spdxElementId")
+        related_spdx_element: str = relationship_dict.get("relatedSpdxElement")
+        relationship_type: Optional[RelationshipType] = parse_field_or_log_error(logger=logger,
+                                                                                 field=relationship_dict.get("relationshipType"),
+                                                                                 parsing_method=self.parse_relationship_type)
+        relationship_comment: str = relationship_dict.get("comment")
+        raise_parsing_error_if_logger_has_messages(logger, "relationship")
+
+        relationship = construct_or_raise_parsing_error(Relationship, dict(spdx_element_id=spdx_element_id,
+                                                                           relationship_type=relationship_type,
+                                                                           related_spdx_element_id=related_spdx_element,
+                                                                           comment=relationship_comment))
+        return relationship
+
+    @staticmethod
+    def parse_relationship_type(relationship_type_str: str) -> RelationshipType:
+        try:
+            relationship_type = RelationshipType[json_str_to_enum_name(relationship_type_str)]
+        except KeyError:
+            raise SPDXParsingError([f"RelationshipType {relationship_type_str} is not valid."])
+        except AttributeError:
+            raise SPDXParsingError([f"RelationshipType must be str, not {type(relationship_type_str).__name__}."])
```

Comment:
No, IIRC I had a failing test which lead to this catch. I deleted this and added an error handling to `json_str_to_enum_name` in case the given argument is not a `str`.